completion/available/git.completion.bash

```diff
@@ -1,3386 +1,24 @@
-#    *) .git/remotes file names
-#    *) git 'subcommands'
-#    *) git email aliases for git-send-email
-#    *) tree paths within 'ref:path/to/file' expressions
-#    *) file paths within current working directory and index
-#    *) common --long-options
-#
-# To use these routines:
-#
-#    1) Copy this file to somewhere (e.g. ~/.git-completion.bash).
-#    2) Add the following line to your .bashrc/.zshrc:
-#        source ~/.git-completion.bash
-#    3) Consider changing your PS1 to also show the current branch,
-#       see git-prompt.sh for details.
-#
-# If you use complex aliases of form '!f() { ... }; f', you can use the null
-# command ':' as the first command in the function body to declare the desired
-# completion style.  For example '!f() { : git commit ; ... }; f' will
-# tell the completion to use commit completion.  This also works with aliases
-# of form "!sh -c '...'".  For example, "!sh -c ': git commit ; ... '".
-#
-# Compatible with bash 3.2.57.
-#
-# You can set the following environment variables to influence the behavior of
-# the completion routines:
-#
-#   GIT_COMPLETION_CHECKOUT_NO_GUESS
-#
-#     When set to "1", do not include "DWIM" suggestions in git-checkout
-#     and git-switch completion (e.g., completing "foo" when "origin/foo"
-#     exists).
+#!/usr/bin/env bash
 
-case "$COMP_WORDBREAKS" in
-*:*) : great ;;
-*)   COMP_WORDBREAKS="$COMP_WORDBREAKS:"
-esac
+# Make sure git is installed
+_command_exists git || return
 
-# Discovers the path to the git repository taking any '--git-dir=<path>' and
-# '-C <path>' options into account and stores it in the $__git_repo_path
-# variable.
-__git_find_repo_path ()
-{
-	if [ -n "$__git_repo_path" ]; then
-		# we already know where it is
-		return
-	fi
+# Don't handle completion if it's already managed
+complete -p git &>/dev/null && return
 
-	if [ -n "${__git_C_args-}" ]; then
-		__git_repo_path="$(git "${__git_C_args[@]}" \
-			${__git_dir:+--git-dir="$__git_dir"} \
-			rev-parse --absolute-git-dir 2>/dev/null)"
-	elif [ -n "${__git_dir-}" ]; then
-		test -d "$__git_dir" &&
-		__git_repo_path="$__git_dir"
-	elif [ -n "${GIT_DIR-}" ]; then
-		test -d "${GIT_DIR-}" &&
-		__git_repo_path="$GIT_DIR"
-	elif [ -d .git ]; then
-		__git_repo_path=.git
-	else
-		__git_repo_path="$(git rev-parse --git-dir 2>/dev/null)"
-	fi
-}
+_git_bash_completion_paths=(
+  # MacOS non-system locations
+  '/Library/Developer/CommandLineTools/usr/share/git-core/git-completion.bash'
+  '/Applications/Xcode.app/Contents/Developer/usr/share/git-core/git-completion.bash'
+)
 
-# Deprecated: use __git_find_repo_path() and $__git_repo_path instead
-# __gitdir accepts 0 or 1 arguments (i.e., location)
-# returns location of .git repo
-__gitdir ()
-{
-	if [ -z "${1-}" ]; then
-		__git_find_repo_path || return 1
-		echo "$__git_repo_path"
-	elif [ -d "$1/.git" ]; then
-		echo "$1/.git"
-	else
-		echo "$1"
-	fi
-}
+for _comp_path in "${_git_bash_completion_paths[@]}" ; do
```

Comment:
Since this is only covering macOS paths now, do we want to wrap it in an OS check `if`?

For Linux, the above paths don't make any sense, no need to check there...

Or have a default empty `_git_bash_completion_paths` array, and only set it to the above values when we're running on macOS.

Reply:
Works for me